### PR TITLE
SC2: Reserve word 16 for raceswap

### PR DESCRIPTION
**Pull request by EnvyDragon**
## What is this fixing or adding?
Just reserving a value in the SetOptions string; the maps' race-swap triggers expect to see either 0 (for no change) or 1/2/3 (for terran/zerg/protoss) on word 16, but the generator logic isn't ready yet, so this will allow that work to be merged and applied without worrying about potential conflicts.

## How was this tested?
1. Compiled the maps with race-swap triggers active
2. Loaded a map that was race-swappable (Smash and Grab); verified that it loaded and ran correctly with a 0 in the race override slot
3. Loaded a map that was NOT race-swappable yet (Zero Hour); verified that it loaded and ran correctly ie. having a value in the slot didn't cause issues
4. Did /download_data to get the version of the maps without race-swap triggers whatsoever; loaded Smash and Grab again, verified that it ran correctly ie. no weird behavior as a result of an extra character in the options string

TLDR having the 0 doesn't break maps that aren't set up to read the value, and correctly produces default behavior on maps that ARE set up to read it

### CODE REVIEW

File: worlds/sc2/client.py [hunk 1]
```diff
@@ -1094,6 +1094,7 @@ async def on_step(self, iteration: int):
             kerrigan_options = calculate_kerrigan_options(self.ctx)
             soa_options = caclulate_soa_options(self.ctx)
             uncollected_objectives: typing.List[int] = self.get_uncollected_objectives()
+            mission_variant = 0  # temp value until generator logic is done
```
Comment (Ziktofel) [inline worlds/sc2/client.py]: Add a TODO comment, also mention race-swapping there with its values, as currently understood by the mod

**Review by Ziktofel — CHANGES_REQUESTED** [review]
